src/lightning/fabric/strategies/xla.py

```diff
@@ -84,6 +84,15 @@ def _configure_launcher(self) -> None:
         self._launcher = _XLALauncher(self)
 
     def setup_environment(self) -> None:
+        from torch_xla.experimental.pjrt import using_pjrt
+
+        if using_pjrt() and len(self.parallel_devices) == 1:
+            # this would raise an internal XLA error. Can be checked with `devices=1, strategy="xla"`
+            raise NotImplementedError(
+                "The `XLAStrategy` does not support running on a single device with the PjRT runtime."
+                " Try using all devices or the `SingleTPUStrategy` strategy"
+            )
```

Comment:
For instance, `ddp_spawn` supports running with a single process.
This is just useful for debugging DDP-specific logic without launching expensive processes. It's not performant.

The expectation would be that `xmp.spawn` supports that too. Seems like it did with `XRT` as we had one test that did it.

If `PjRT` does not support it, it's okay. I added this check to avoid footguns. But I'd like to know if it's expected or a bug

Reply:
I see, I don't think it is supported.